sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -71,22 +71,14 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
   }
 
   override def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
-    val stopEarly =
-      ctx.addMutableState(CodeGenerator.JAVA_BOOLEAN, "stopEarly") // init as stopEarly = false
-
-    ctx.addNewFunction("stopEarly", s"""
-      @Override
-      protected boolean stopEarly() {
-        return $stopEarly;
-      }
-    """, inlineToOuterClass = true)
     val countTerm = ctx.addMutableState(CodeGenerator.JAVA_INT, "count") // init as count = 0
     s"""
        | if ($countTerm < $limit) {
        |   $countTerm += 1;
+       |   if ($countTerm == $limit) {
+       |     setStopEarly(true);
```

Comment:
shall we do this after `consume`?

Reply:
won't we call `shouldStop` inside `consume`? if it does, `stopEarly` will not be set.

Reply:
`if ($countTerm == $limit)` means this is the last record, and we should still consume it?

Reply:
Oh, I see. And I think `shouldStop` shouldn't be called inside `consume`.